src/modules/export.js

```diff
@@ -379,13 +399,9 @@ export const startExport = () => {
       : exportData.query;
 
     // filter out only the fields we want to include in our export data
-    const projection = Object.keys(exportData.fields)
-      .filter(field => exportData.fields[field] === 1)
-      .reduce((obj, field) => {
-        obj[field] = exportData.fields[field];
-
-        return obj;
-      }, {});
+    const projection = Object.fromEntries(
+      Object.entries(exportData.fields)
+        .filter((keyAndValue) => keyAndValue[1] === 1));
 
     dataService.estimatedCount(ns, {query: spec.filter}, function(countErr, numDocsToExport) {
```

Comment:
How is this working with a query? :/ The dataService uses this: (http://mongodb.github.io/node-mongodb-native/3.6/api/Collection.html#estimatedDocumentCount), we probably report a very strange progress :)

On top of that i think a user could have set a limit from the query bar, which we seem to ignore.

Reply:
I think it’s fine to open a ticket about this if we can improve it, but it seems to be working well enough for me, and I think it’s also okay for progress bars to be inaccurate sometimes :)